### ISSUE TITLE
Name of the report is not saving as expected (even though 'reportFilename' in cypress.json file is changed)

### ISSUE THREAD

**Issue by prudhvidandamudi** [adamgruber/mochawesome#293]
i am using mochawesome for my cypress tests. 

**This is my cypress.json file**
```
{
  "viewportWidth": 1440,
  "viewportHeight": 800,
  "defaultCommandTimeout": 8000,
  "reporter": "mochawesome",
  "reporterOptions": {
      "reportDir": "cypress/reports/mochawesome-report",
      "reportFilename": "sample",
      "overwrite": false,
      "html": true,
      "json": true  
   },
  "env": {
    "environment": "Stagging"
  },
  "screenshotsFolder": "cypress/reports/mochawesome-report/assets",
}
```

when I ran my tests, the reports are generated with name '**sample.html**' which is expected as I have defined my report title as sample in **cypress.json** file.

now I wanted to have unique name for each report generated based on the test suite name.

so I have used the **cypress event** '`test:before:run`' and set the `reportFilename` in **cypress.json** to the test suite name before the execution of each test, the sample code is down below

```
Cypress.on("test:before:run", (test, runnable) => {
    console.log(Cypress.config("reporterOptions").reportFilename);
    const suiteTitle = runnable.parent.title;
    Cypress.config({
        "reporterOptions":{
            "reportFilename": suiteTitle
        }
    });
    const reportFileName = Cypress.config("reporterOptions").reportFilename;
    console.log(reportFileName);
});
```
I see the `reportFilename` is successfully changed to Tests suite name 

![image](https://user-images.githubusercontent.com/46937269/67519899-cf3d0500-f675-11e9-8e26-385f69fac00c.png)

**But my report is still saved as 'sample.html'**
![image](https://user-images.githubusercontent.com/46937269/67519965-ee3b9700-f675-11e9-9732-167051cc8db5.png)

**Why my report is still saving as sample.html , instead of my test suite name?**

can you help me to resolve this @adamgruber ?

**Comment by adamgruber** [comment]
This will not work because the reporter has already been created with the initial options and no longer references the Cypress config. There is currently no way to have custom filenames for each test.

**Comment by prudhvidandamudi** [comment]
@adamgruber 

Is it possible to implement it as a new feature?
So that each report file is created with the name of the Spec files by default
 or 
at least an option to alter the reporter options before test execution, so that file be saved under the altered name?

**Comment by adamgruber** [comment]
It might be possible but I don't have a ton of time to devote to it right now.

**Comment by adamgruber** [comment]
Looking into this a bit, adding the feature is pretty straightforward but to work with Cypress will require https://github.com/cypress-io/cypress/issues/1495 to be complete.

**Comment by mohanhbk** [comment]
is this feature implemented?

**Comment by adamgruber** [comment]
Not yet but PRs are welcome.

**Comment by adamgruber** [comment]
Please see https://github.com/adamgruber/mochawesome#reportfilename-replacement-tokens which is now available in v7.1.0. I believe this should address your issues.
